consensus/service/src/tx_manager/mod.rs

```diff
@@ -105,6 +105,38 @@ impl<E: ConsensusEnclave + Send, UI: UntrustedInterfaces + Send> TxManagerImpl<E
     fn lock_cache(&self) -> MutexGuard<HashMap<TxHash, CacheEntry>> {
         self.cache.lock().expect("Lock poisoned")
     }
+
+    /// A utility method for resolving a list of TxHashes into CacheEntries that
+    /// errors if any hashes are missing.
+    fn get_cache_entries<'a, 'b, I>(
+        cache: &'a MutexGuard<HashMap<TxHash, CacheEntry>>,
```

Comment:
I am passing the locked cache instead of `&self` and locking inside so that the caller has control over the shared access to the cache.